common/models/user.js

```diff
@@ -438,6 +438,65 @@ module.exports = function(User) {
   };
 
   /**
+   * Generates a temporary passoword reset access token and triggers the requestPasswordReset event.
+   *
+   * ```js
+   *    var options = {
+   *      to: user.email,
+   *      generateVerificationToken: function (user, cb) { cb(err , "random-token"); }
+   *    };
+   *
+   *    user.requestPasswordReset(options, next);
+   * ```
+   *
+   * @options {Object} options
+   * @property {String} to Email address to which verification email is sent.
+   * @property {Function} generateVerificationToken A function to be used to
+   *  generate the verification token. It must accept the user object and a
+   *  callback function. This function should NOT add the token to the user
+   *  object, instead simply execute the callback with the token! User saving
+   *  and email sending will be handled in the `verify()` method.
+   */
+
+  User.prototype.requestPasswordReset = function(options, fn) {
+    fn = fn || utils.createPromiseCallback();
+    options = options || {};
+
+    var user = this;
+    var userModel = this.constructor;
+    var ttl = userModel.settings.resetPasswordTokenTTL || DEFAULT_RESET_PW_TTL;
+
+    // Set the email to the user if it is not set
+    options.to = options.to || user.email;
+
+    // Set default token generator function if one is not provided
+    var tokenGenerator = options.generateVerificationToken || userModel.generateVerificationToken;
```

Comment:
So far, I believe we were not allowing user-provided `options` to provide functions to customize the behaviour. One of the reasons is that such functions cannot be specified via REST API.

Instead, we are expecting applications to subclass the built-in `User` model and change the behaviour by overriding certain functions, for example `User.prototype.createAccessToken`.

I would prefer to keep that approach, unless you have a specific need for providing different `tokenGenerator` functions on per-request basis.

Thoughts?